src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -726,6 +737,36 @@ bool InputStateMachineEngine::_WriteSingleKey(const short vkey, const DWORD dwMo
     return _WriteSingleKey(wch, vkey, dwModifierState);
 }
 
+// Method Description:
+// - Writes a Mouse Event Record to the input callback based on the state of the mouse.
+// Arguments:
+// - uiColumn - the X/Column position on the viewport (0 = left-most)
+// - uiLine - the Y/Line/Row position on the viewport (0 = top-most)
+// - dwButtonState - the mouse buttons that are being modified
+// - dwModifierState - the modifier state to write mouse record.
+// - dwEventFlags - the type of mouse event to write to the mouse record.
+// Return Value:
+// - true iff we successfully wrote the keypress to the input callback.
+bool InputStateMachineEngine::_WriteMouseEvent(const size_t column, const size_t line, const DWORD buttonState, const DWORD controlKeyState, const DWORD eventFlags)
+{
+    // At most 1 record - the modifiers don't get their own events
+    const size_t MAX_LENGTH = 1;
+    INPUT_RECORD rgInput[MAX_LENGTH];
+    size_t cInput = MAX_LENGTH;
+
+    COORD uiPos = { gsl::narrow<short>(column) - 1, gsl::narrow<short>(line) - 1 };
+
+    rgInput[0].EventType = MOUSE_EVENT;
+    rgInput[0].Event.MouseEvent.dwMousePosition = uiPos;
+    rgInput[0].Event.MouseEvent.dwButtonState = buttonState;
+    rgInput[0].Event.MouseEvent.dwControlKeyState = controlKeyState;
+    rgInput[0].Event.MouseEvent.dwEventFlags = eventFlags;
+
+    // pack and write input records
+    std::deque<std::unique_ptr<IInputEvent>> inputEvents = IInputEvent::Create(gsl::make_span(rgInput, cInput));
+    return _pDispatch->WriteInput(inputEvents);

```

Comment:
I don't really understand why this is a 1-size array. Can't it just be a single `INPUT_RECORD` and then do `gsl::make_span(&record, 1)`?